SSA/Projects/LLVMRiscV/Pipeline/tactics/SimpLowering.lean

```diff
@@ -0,0 +1,54 @@
+import SSA.Projects.RISCV64.Tactic.SimpRiscV
+import SSA.Projects.InstCombine.ForLean
+import SSA.Projects.InstCombine.Tactic
+import SSA.Projects.InstCombine.TacticAuto
+import SSA.Projects.InstCombine.LLVM.EDSL
+import Lean.Meta.Tactic.Simp.SimpTheorems
+import Lean.Meta.Tactic.Simp.RegisterCommand
+import Lean.LabelAttribute
+
+
+open Lean
+open Lean.Elab.Tactic
+
+/-!
+This file defines the `simp_lowering` tactic, which bundles the llvm tactics as well as the riscv simp tactic
+and invokes bv_decide. The tactic is engineered to solve any instruction lowering patterns within the
+domain solveable by bv_decide. We enahnced the llvm tactics to handle the Poison cases and the riscv tactic
+simplifies riscv defintions such that the remaining statement is solveable by bv_decide.
+It also defines the `simp_lowering` attribute, which can be used to tag semantic definition functions so that
+they are automatically simplified when `simp_lowering` is invoked. Moreover to achieve this proof automation,
+a rewriter implementor must tag its rewrites as @[simp_denote] such that the first innovcation of `simp_peephole`
+within the `simp_lowering` tactic unfolds the defintions to allow further tactics to be applied.
+
+The purpose of this tactic is to reduce proof size when reasoning about instruction lowerings.
+
+/-
+This file defines the `simp_lowering` tactic, which combines the LLVM tactics with the RISC-V simplification
+tactic and invokes `bv_decide`. The tactic is designed to solve instruction lowering patterns that
+fall within the domain solvable by `bv_decide`. We have enhanced the LLVM tactics to handle PoisonOr
+cases, and the RISC-V tactic simplifies RISC-V definitions so that the remaining goals are suitable
+for `bv_decide`.
+It also defines the `simp_lowering` attribute, which can be used to tag semantic definition
+functions so that they are automatically simplified when `simp_lowering` is invoked.
+To enable this proof automation, rewriter implementors must tag their rewrite rules with
+`@[simp_denote]`. This ensures that the initial invocation of `simp_peephole` within the
+`simp_lowering` tactic unfolds the relevant definitions, allowing subsequent tactics to proceed.
+
+The purpose of this tactic is to maximize proof automation for insstruction lowering rewrite patterns
+-/
+-/
+
+syntax "simp_lowering" : tactic
+macro_rules
+| `(tactic| simp_lowering) =>
+  `(tactic|(
+    simp_peephole
+    simp_alive_undef
+    simp_riscv
+    simp_alive_ops
+    simp_alive_case_bash
+    simp_alive_split
+    try all_goals bv_decide --only run `bv_decide` if there are goals left
```

Comment:
```suggestion
    simp_peephole
    simp_alive_undef
    simp_riscv
    simp_alive_ops
    simp_alive_case_bash
    simp_alive_split
```

I would not put bv_decide here, so that the tactic is still reliably usable in situations where bv_decide doesn't work.
Instead, you have the simp_lowering tactic call only the simp_ tactics, and then in the auto-tactic you write simp_lowering <;> bv_decide.